### ISSUE TITLE
QuickLook Previews Never Finish on M1 Mac and Big Sur

### ISSUE THREAD

**Issue by alexclst** [sbarex/SourceCodeSyntaxHighlight#51]
On a Mac with the M1 chip and Big Sur the previews in QuickLook just spin endlessly, and never actually finish showing a preview. My guess is that something is hanging with `highlight`, likely as a side-effect of the new architecture. I know the app is running under Rosetta 2, so that may be a contributing factor as well.

**Comment by sbarex** [comment]
Yes, the application is not yet universal and nor `highlight`.
On Catalina I cannot compile code for M1 chip (why?). I need to upgrade os in the next weeks.

**Comment by alexclst** [comment]
Neither output mode seems to work. RTF does show preview in the main app, but fails in QuickLook.

**Comment by sbarex** [comment]
Ok, I must study if big sur on m1 can handle extension compiled for intel...
I hope to add M1 support in the next release (if `highlight` and `lua` can be compiled for M1 wihout edit the source).

**Comment by sbarex** [comment]
Please try the new release. I have compiled the app and `highlight` as universal binary, so I hope works as native application on M1.
Please notify me if there are still problems. I have not yet had the opportunity to test it on Big Sur (not even on an Intel platform)

**Comment by alexclst** [comment]
It appears to be stuck in the exact same way, despite being fully native. Maybe there is another issue with `highlight` or the QuickLook extension in Big Sur that is the problem regardless of system architecture?

**Comment by sbarex** [comment]
This weekend I hope to be able to install Big Sur and test there.

Running in native mode, the main app works? Just to understand if the connection with the `highligth` process works.

Also, enable the debug option (on the global settings). Then when you try to generate a quicklook preview, on your desktop will be created two files (`colorize.log` and `colorize.rtf` or `colorize.html`). Can you post the `colorize.log` file?

**Comment by alexclst** [comment]
Yes, the main app is working fine. So maybe an issue with the extension. Colorize log: 
[colorize.log](https://github.com/sbarex/SourceCodeSyntaxHighlight/files/5592362/colorize.log)
Testing the same build of the app on an Intel Big Sur Mac it works just fine.

**Comment by sbarex** [comment]
Thanks for the log file! I see that the the ql exetension call correctly `highlight`. From this I can say however that it does not seem to me to be a problem with the quicklook system architecture.
If the file `colorize.html` contains formatted code then the problem is certainly in the preview window.
In the settings, if you try to set the output to rtf the preview works?

**Comment by alexclst** [comment]
That method works. Tried that with the earlier build, but not this latest build, I guess. Didn't work then. Works now. But, that still means 50% of the way someone may set up their preview, and the default mode at that, is totally failing on M1...

**Comment by sbarex** [comment]
The `.html` format seems to be giving problems to more people and it is something I have to try to solve. Meanwhile now the output for those who have not previously set preferences is the `.rtf` format.

For differences between `.html` and `.rtf` output you can see this [post](https://github.com/sbarex/SourceCodeSyntaxHighlight/issues/48#issuecomment-729571030).

In the meantime, thanks for the checks on the M1 chip (that I am beginning to envy)!

**Comment by alexclst** [comment]
Feel free to ask me to do more as needed... I'm comfortable within Xcode too, so can definitely help debug more as needed. Although, incidentally, I could never get the current `master` branch to build on my Mac today, it kept failing saying something about Lua not being available, even after I updated the submodule. I tried to build it, because I half thought that maybe since the M1 and Big Sur are stricter about code signing that maybe a self-built copy would work.

**Comment by sbarex** [comment]
Thanks for the support. I think it may be a problem on the grants to run webkit, perhaps in the entitlements. It took me some time to understand that to use webkit you had to set `com.apple.security.network.client` to true...

I am not an xcode programming expert. Maybe I have something wrong with setting up the git modules.
I tried to clone the repository and it actually gave me problems with lua. Try to select the target `Lua macOs` and only after applying. This is how it should work.

**Comment by sbarex** [comment]
I have upgraded the computer to big sur. Using the html output the webiview crash with this output:
```
Syntax Highlight Quicklook Extension[13739:455301] [Process] 0x7f8cbc029620 - [pageProxyID=5, webPageID=6, PID=0] WebPageProxy::processDidTerminate: (pid 0), reason 3

Syntax Highlight Quicklook Extension[13739:455301] [Loading] 0x7f8cbc029620 - [pageProxyID=5, webPageID=6, PID=0] WebPageProxy::dispatchProcessDidTerminate: reason = 3

Syntax Highlight Quicklook Extension[13739:455301] [Process] 0x7f8cbc029620 - [pageProxyID=5, webPageID=6, PID=0] WebPageProxy::tryReloadAfterProcessTermination: process crashed and the client did not handle it, not reloading the page because we reached the maximum number of attempts
```

At the moment I can't understand the reasons.
I also tried [glance](https://github.com/samuelmeuli/glance) and it has the same problem.

**Comment by sbarex** [comment]
see this [specific issue for html output](https://github.com/sbarex/SourceCodeSyntaxHighlight/issues/52)

**Comment by sbarex** [comment]
See [rel 1.0.b29](https://github.com/sbarex/SourceCodeSyntaxHighlight/releases/tag/1.0.b29) for a temporary fix.
